plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestExasolTypeMapping.java

```diff
@@ -279,9 +281,110 @@ private void testDate(ZoneId sessionZone)
     void testHashtype()
     {
         SqlDataTypeTest.create()
-                .addRoundTrip("hashtype", "NULL", createCharType(32), "CAST(NULL AS char(32))")
-                .addRoundTrip("hashtype", "'550e8400-e29b-11d4-a716-446655440000'", createCharType(32), "CAST('550e8400e29b11d4a716446655440000' AS char(32))")
-                .execute(getQueryRunner(), exasolCreateAndInsert(TEST_SCHEMA + "." + "test_hashtype"));
+                // Null
+                .addRoundTrip("hashtype", "NULL", VARBINARY, "from_hex(NULL)")
+
+                // Explicit 16-byte
+                .addRoundTrip("hashtype(16 byte)", "'550e8400-e29b-11d4-a716-446655440000'", VARBINARY, "from_hex('550e8400e29b11d4a716446655440000')")
+
+                // Explicit 32-byte
+                .addRoundTrip("hashtype(32 byte)", "'00112233-44556677-8899AABB-CCDDEEFF-00112233-44556677-8899AABB-CCDDEEFF'", VARBINARY,
+                        "from_hex('00112233445566778899AABBCCDDEEFF00112233445566778899AABBCCDDEEFF')")
+                .addRoundTrip("hashtype(32 byte)", "'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF'", VARBINARY,
+                        "from_hex('FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF')")
+
+                // Boundary: minimum size (1 byte)
+                .addRoundTrip("hashtype(1 byte)", "'FF'", VARBINARY, "from_hex('FF')")
+
+                // Boundary: maximum size (1024 byte) - use repeated pattern
+                .addRoundTrip("hashtype(1024 byte)", "'" + "AA".repeat(1024) + "'", VARBINARY, "from_hex('" + "AA".repeat(1024) + "')")
+
+                .execute(getQueryRunner(), exasolCreateAndInsert(TEST_SCHEMA + "." + "test_hashtype_as_varbinary_mapping"));
+    }
+
+    @Test
+    // See for more details: https://docs.exasol.com/saas/microcontent/Resources/MicroContent/general/hash-data-type.htm
```

Comment:
Move this comment above `@Test`. Otherwise, auto-format changes the indentation.